### PR TITLE
Jolive3000 5.5 custom maze testing

### PR DESCRIPTION
**Pull request by jolive3000**
Tests and some CA improvements for the custom maze section

Editing to add a summary in response to Seamus' comment:
- fixed the problems mentioned in the rubric (SOLID and CA violations) mainly by moving 3 adapter layer classes so one became and entity (EditorTile) and the other two use cases (TempMaze & EditorOnClick) 
- Boundaries were added for classes that depended on the above and improper dependancy was fixed in a few places
- Implemented Seamus' MazeInfo (formerly Maze) for retrieving of constants
- Added tests classes for CustomGameFileManager, EditorTile and TempMaze
- Made the player's start location fixed

### CODE REVIEW

File: src/main/java/entities/custom_game/EditorTile.java [hunk 1]
```diff
@@ -11,33 +13,19 @@
  * EditorTiles are also stored in TempMaze for conversion to a text file (storage)
  */
 public class EditorTile extends JLabel {
-    private final int X;
-    private final int Y;
     private String name;
     private int numCode;
-
-    private static final String[] secondaryMenuItems = {"photons", "key", "enemy", "start", "end"};
+    private static final String[] secondaryMenuItems = {"photons", "key", "stationaryEnemy", "chasingEnemy", "start", "end"};
```
Comment (tseamuscorlett) [inline src/main/java/entities/custom_game/EditorTile.java:18]: In my most recent PR, I changed all mentions of "Photons" to "Oxygen" and added a new .png file.

**Comment by jolive3000** [comment]
Currently, start location is editable but I can make it fixed to position 1, 1 easily in the next pull request

**Comment by tseamuscorlett** [comment]
> Currently, start location is editable but I can make it fixed to position 1, 1 easily in the next pull request

AssetSetter doesn't support variable Player start position, so fixed position would be better!

**Comment by tseamuscorlett** [comment]
> Tests and some CA improvements for the custom maze section


You should explain more what's contained in this PR

**Review by enerko — APPROVED** [review]
Awesome! Checked for Javadoc, CA, tests, code style, functionality and all seems to be good! I was just wondering if we're going to be able to edit the mazes since there was a comment on this in the rubric. If we are not going to be able to view/edit the mazes, then I think we should get rid of the file list in the main frame completely. It would be great if an instruction on how to make a maze is added to the main frame, so that the user know how to place objects, etc. 

These things can be addressed in another PR, I think this one should be merged into main before it becomes any longer.

**Pull request merged**